### PR TITLE
Add support for python 3.10

### PR DESCRIPTION
**Pull request by ancalita**
**Proposed changes**:
- Update `pyproject.toml` and CI workflow.
- Remove documentation workflow since sdk docs now live in the rasa-oss repo.

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa-sdk/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa-sdk#code-style) for instructions)

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -24,10 +24,10 @@ jobs:
       - name: Checkout git repository 🕝
         uses: actions/checkout@v2
 
-      - name: Set up Python 3.7 🐍
-        uses: actions/setup-python@v1
+      - name: Set up Python 3.10 🐍
+        uses: actions/setup-python@b55428b1882923874294fa556849718a1d7f2ca5
         with:
-          python-version: 3.7
+          python-version: 3.10.8
```
Comment (m-vdb) [inline .github/workflows/continuous-integration.yml:30]: why pin this specific version instead of relying on the latest micro available?

**Review by m-vdb — APPROVED** [review]
💯

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (ancalita) [reply .github/workflows/continuous-integration.yml:30]: The CI seems to read `3.10` as `3.1`, like [here](https://github.com/RasaHQ/rasa-sdk/actions/runs/3496484220/jobs/5854446771#step:3:3), and I didn't know another alternative to fix this.

Comment (ancalita) [reply .github/workflows/continuous-integration.yml:30]: A quick search online seems to suggest to wrap this in single quotes as '3.10' instead, I'll try that instead 🤞🏻

Comment (m-vdb) [reply .github/workflows/continuous-integration.yml]: yep

**Comment by ancalita** [comment]
@miraai Could I please get your review on the workflow changes? 🙏🏻

**Review by m-vdb — APPROVED** [review]
💯 💯

**Review by miraai — APPROVED** [review]
🚀

**Comment by gabrielhjs** [comment]
Hey, @ancalita! Any predictions for a docker image with python 3.10.x?

**Comment by ancalita** [comment]
@gabrielhjs Thanks for flagging this, I overlooked updating the base image to `ubuntu:22.04` in the `Dockerfile` 🙈 
This change has been made on the `3.4.x` branch, and I've opened a [PR](https://github.com/RasaHQ/rasa-sdk/pull/946) to merge this change in `main` too. At the moment, there are no changes that could be released in a `3.4.x` micro, however the upcoming minor `3.5.0` will definitely have an updated Docker image. Apologies again for the issue!

**Pull request merged**